File: vortexdbg-api/src/main/java/com/vortexdbg/arm/context/Arm64RegisterContext.java
package com.vortexdbg.arm.context;

import com.vortexdbg.pointer.VortexdbgPointer;

public interface Arm64RegisterContext extends RegisterContext {

    long getXLong(int index);

    int getXInt(int index);

    VortexdbgPointer getXPointer(int index);

    long getFp();

    VortexdbgPointer getFpPointer();

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/arm/context/Arm64RegisterContext.kt
package com.vortexdbg.arm.context

import com.vortexdbg.pointer.VortexdbgPointer

interface Arm64RegisterContext : RegisterContext {

    fun getXLong(index: Int): Long

    fun getXInt(index: Int): Int

    fun getXPointer(index: Int): VortexdbgPointer

    fun getFp(): Long

    fun getFpPointer(): VortexdbgPointer

}
